clients/src/main/java/org/apache/kafka/clients/producer/internals/Sender.java

```diff
@@ -582,25 +584,27 @@ private void handleProduceResponse(ClientResponse response, Map<TopicPartition,
                 requestHeader, response.destination());
             for (ProducerBatch batch : batches.values())
                 completeBatch(batch, new ProduceResponse.PartitionResponse(Errors.REQUEST_TIMED_OUT, String.format("Disconnected from node %s due to timeout", response.destination())),
-                        correlationId, now);
+                        correlationId, now, new HashMap<>());
```

Comment:
It does seem wasteful constructing an unnecessary new HashMap in all these cases.